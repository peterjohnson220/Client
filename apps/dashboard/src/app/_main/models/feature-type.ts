import { FeatureTypes } from './feature-types';

export class FeatureType {
  AllTypes: FeatureTypes[] = [];

  Activity: FeatureTypes.Activity;
  Community: FeatureTypes.Community;
  DataInsights: FeatureTypes.DataInsights;
  Employees: FeatureTypes.Employees;
  JobDescriptions: FeatureTypes.JobDescriptions;
  MyJobs: FeatureTypes.MyJobs;
  PayMarkets: FeatureTypes.PayMarkets;
  PricingProjects: FeatureTypes.PricingProjects;
  Resources: FeatureTypes.Resources;
  Service: FeatureTypes.Service;
  Structures: FeatureTypes.Structures;
  Surveys: FeatureTypes.Surveys;

  constructor() {
    this.AllTypes = [
      FeatureTypes.Activity,
      FeatureTypes.Community,
      FeatureTypes.DataInsights,
      FeatureTypes.Employees,
      FeatureTypes.JobDescriptions,
      FeatureTypes.MyJobs,
      FeatureTypes.PayMarkets,
      FeatureTypes.PricingProjects,
      FeatureTypes.Resources,
      FeatureTypes.Service,
      FeatureTypes.Structures,
      FeatureTypes.Surveys
    ];
  }
}
