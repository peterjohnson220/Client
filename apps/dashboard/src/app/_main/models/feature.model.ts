import { FeatureTypes } from './feature-types';
import { TileTypes } from './tile-types';

export interface Feature {
  Type: FeatureTypes;
  TileType: TileTypes;
  HasAccess: boolean;
  Url: string;
}
