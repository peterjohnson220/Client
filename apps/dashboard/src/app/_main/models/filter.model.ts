export interface TimelineActivityFilter {
  Label: string;
  Value: string;
  IsEnabled: boolean;
}
