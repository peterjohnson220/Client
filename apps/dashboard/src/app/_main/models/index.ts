export * from './timeline-activity.model';
