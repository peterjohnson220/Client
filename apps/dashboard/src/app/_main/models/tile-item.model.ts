import { Type } from '@angular/core';

export class TileItem {
  constructor(public component: Type<any>, public data: any) {}
}
