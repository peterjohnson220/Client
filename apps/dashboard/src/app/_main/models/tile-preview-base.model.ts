export interface TilePreviewBase {
  IconClass?: string;
}
