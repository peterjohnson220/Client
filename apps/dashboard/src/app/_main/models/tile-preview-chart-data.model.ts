export interface TilePreviewChartData {
  CategoryName: string;
  CategoryValue: number;
  DetailData?: { Key: string, Value: number }[];
}
export function generateMockTilePreviewChartData(categoryName: string,
                                             categoryValue: number = 0,
                                             detailKey: string = '',
                                             detailValue: number = 0): TilePreviewChartData {
    return {
      CategoryName: categoryName,
      CategoryValue: categoryValue,
      DetailData: [ { Key: detailKey, Value: detailValue } ]
    };
}
