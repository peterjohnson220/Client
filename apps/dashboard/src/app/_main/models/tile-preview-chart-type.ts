import { TilePreviewChartTypes } from './tile-preview-chart-types';

export class TilePreviewChartType {
  AllTypes: TilePreviewChartTypes[];

  Donut = TilePreviewChartTypes.Donut;
  Pie = TilePreviewChartTypes.Pie;

  constructor() {
    this.AllTypes = [
      TilePreviewChartTypes.Donut,
      TilePreviewChartTypes.Pie
    ];
  }
}
