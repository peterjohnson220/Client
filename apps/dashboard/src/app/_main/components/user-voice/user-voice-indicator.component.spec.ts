import { NO_ERRORS_SCHEMA } from '@angular/core';
import { TestBed, ComponentFixture } from '@angular/core/testing';

import { UserVoiceIndicatorComponent } from './user-voice-indicator.component';
import { generateMockUserVoiceModel } from '../../models';

describe('User Voice Indicator', () => {
  let fixture: ComponentFixture<UserVoiceIndicatorComponent>;
  let instance: UserVoiceIndicatorComponent;

  // Configure Testing Module for before each test
  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [
        UserVoiceIndicatorComponent
      ],
      // Shallow Testing
      schemas: [ NO_ERRORS_SCHEMA ]
    });

    fixture = TestBed.createComponent(UserVoiceIndicatorComponent);
    instance = fixture.componentInstance;
  });

  it('should show the link to user voice', () => {

    instance.model = generateMockUserVoiceModel();

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
