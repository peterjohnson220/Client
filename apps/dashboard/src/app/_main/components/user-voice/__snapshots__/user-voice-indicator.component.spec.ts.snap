// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Voice Indicator should show the link to user voice 1`] = `
<pf-user-voice-indicator
  userContext={[Function Object]}
  userVoiceForumId={[Function Number]}
  userVoiceUrl={[Function String]}
>
  <div
    class="user-voice-container"
  >
    
    
    <a
      href="/payfactors/UserVoice/Redirect?userId=1&url=http://payfactors.uservoice.com/forums/268517"
    >
      
      
      <span
        class="fa-stack fa-2x"
      >
        
        
        <i
          class="fa fa-circle fa-stack-2x user-voice-icon-color"
        />
        
        
        <i
          class="fa fa-circle-thin fa-stack-2x text-white"
        />
        
        
        <i
          class="fa fa-lightbulb-o fa-stack-1x text-white"
        />
        
      
      </span>
      
    
    </a>
    
  
  </div>
  
</pf-user-voice-indicator>
`;
